### ISSUE TITLE
Window should escape container when it's alone and I try to move it.

### ISSUE THREAD

**Issue by WhyNotHugo** [swaywm/sway#5644]
This isn't new behaviour, it was already in i3 last year.

## Steps to reproduce
  - Open a tabbed container (let call this `parentContainer`) with two children (who are also containers):
    - Open two windows, vertically split, inside the first one.
    - Open two windows, vertically split, inside the second one (let's call this `rightContainer`).
  - focus on one of the windows on the left.
  - move it right

### Expected behaviour (i3)
 - The window becomes a child of `container1`, and a sibling of its former parent.
 - In other words, `container1` now has three children.

### Actual behaviour (sway)
 - The window becomes a child of `rightContainer`.
 
Because of this subtle different, getting a window "out" of a mix of nested containers get far more complicated. And I _do_ tend to nest a lot.

## Finer steps to reproduce
- Open a window
- `split v`
- Open a window
- `layout tabbed`
- (focus should be on a window)
- `split v`
- Open a window
- `focus left`
- `split v`
- Open a window
- `move right` <<-- this is the step that behaves differently.

**Comment by Xyene** [comment]
Does https://github.com/swaywm/sway/pull/5709 fix this for you? Shot in the dark, but nonetheless a related issue.

I wasn't entirely able to understand your reproduction steps (which of the windows is `container1` -- the reproduction steps only mention `parentContainer` and `rightContainer`?)

**Comment by WhyNotHugo** [comment]
This is my starting state:

![2020-09-26T22:39:28,403791529+02:00](https://user-images.githubusercontent.com/730811/94349906-ab7a0000-0038-11eb-8a6a-abeb824d9938.png)

If I use "move right", swaywm does this (e.g.: I've bound this to <kbd>Super</kbd>+<kbd>l</kbd>:

![2020-09-26T22:40:30,116021720+02:00](https://user-images.githubusercontent.com/730811/94349913-b339a480-0038-11eb-99df-9c67ac3240e3.png)

On i3, it would do this (it's **really** hard to transition to this layout in swaywm!):

![2020-09-26T22:43:01,590252528+02:00](https://user-images.githubusercontent.com/730811/94349943-ed0aab00-0038-11eb-8ed8-714b666eba3a.png)

**Comment by Xyene** [comment]
Ah, I get it now. You have `H[T[app1 app2] T[app3 app4]]`, and when `app2` is focused and `move right` executed, we should be seeing `H[app1 app2 T[app3 app4]]`, but instead we see `H[app1 T[app2 app3 app4]]`.

I assume that `move right` a second time would result in `H[app1 T[app2 app3 app4]]` as well, right?

**Comment by WhyNotHugo** [comment]
(#5709 does not fix it (took a while to figure out how to build it!).

**Comment by WhyNotHugo** [comment]
> I assume that move right a second time would result in H[app1 T[app2 app3 app4]] as well, right?

Yes, i3 also did that! It helps a lot breaking windows out of containers when they're alone, and transitioning through very complex layouts).

**Comment by Xyene** [comment]
I can see why the behavior you describe would be useful, but how is your i3 configured? I just tried this in i3-gaps master, and the resulting layout matches sway's current behavior.

**Comment by WhyNotHugo** [comment]
[This](https://github.com/WhyNotHugo/dotfiles/blob/f241540dc6b8ba1d93009dd768277e5a2e9ce2e2/home/.config/i3/config) was my i3 config last I used it.
